core/src/test/java/io/jenkins/pluginhealth/scoring/probes/ThirdPartyRepositoryDetectionProbeTest.java

```diff
@@ -0,0 +1,61 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+
+import java.io.*;
+import java.util.List;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Repository;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.junit.jupiter.api.Test;
+
+public class ThirdPartyRepositoryDetectionProbeTest extends AbstractProbeTest<ThirdPartyRepositoryDetectionProbe> {
+    @Override
+    ThirdPartyRepositoryDetectionProbe getSpy() {
+        return spy(ThirdPartyRepositoryDetectionProbe.class);
+    }
+
+    @Test
+    void shouldFailIfThirdPartRepositoriesDetected() throws IOException, XmlPullParserException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream("pom-test-both-paths/pom.xml");
+
+        MavenXpp3Reader mavenReader = mock(MavenXpp3Reader.class);
+        Model mockModel = mock(Model.class);
+        List<Repository> repositoryList = mavenReader.read(inputStream).getRepositories();
+        when(mockModel.getRepositories()).thenReturn(repositoryList);
+
+        final ThirdPartyRepositoryDetectionProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "message", "status")
+            .isEqualTo(ProbeResult.failure(ThirdPartyRepositoryDetectionProbe.KEY, "Third party repository detected in the plugin"));
+    }
```

Comment:
This should be repeated for the other invalid `pom.xml` that you created. 
You can see https://github.com/jenkins-infra/plugin-health-scoring/blob/main/core/src/test/java/io/jenkins/pluginhealth/scoring/scores/PluginMaintenanceScoringTest.java about using `@ParameterizedTest`. See https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests

Reply:
I will update this.